From 9fc3d8c283b2ae6d582e10057b8424135f38661b Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Mon, 1 Nov 2021 10:08:08 +0100
Subject: [PATCH 3/8] scripts/mk/buildopts.mk: parse DEB_BUILD_OPTIONS once

Assign DEB_BUILD_OPTION_PARALLEL with := so that DEB_BUILD_OPTIONS is
not parsed every time DEB_BUILD_OPTION_PARALLEL is expanded.
---
 scripts/mk/buildopts.mk | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/scripts/mk/buildopts.mk b/scripts/mk/buildopts.mk
index 6787da76f..dd0cef852 100644
--- a/scripts/mk/buildopts.mk
+++ b/scripts/mk/buildopts.mk
@@ -4,12 +4,16 @@
 # Defines the following variables:
 #
 #   DEB_BUILD_OPTION_PARALLEL: the argument for the parallel=N option.
+#     Examples (the expansion is empty if parralel=N is not given):
+#     # $(MAKE) $(addprefix -j,$(DEB_BUILD_OPTION_PARALLEL))
+#     # SPHINXDOC += $(addprefix -j,$(DEB_BUILD_OPTION_PARALLEL))
 
 ifndef dpkg_buildopts.mk_included
 dpkg_buildopts.mk_included :=
 
-ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
-  DEB_BUILD_OPTION_PARALLEL = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+dpkg_buildopts.mk_parallel := $(filter parallel=%,$(DEB_BUILD_OPTIONS))
+ifneq (,$(dpkg_buildopts.mk_parallel))
+  DEB_BUILD_OPTIONS_PARALLEL := $(dpkg_buildopts.mk_parallel:parallel=%,%)
 endif
 
 endif
-- 
2.39.2

